src/Features/CSharp/Portable/UseNameOf/CSharpUseNameofCodeFixProvider.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace Microsoft.CodeAnalysis.CSharp.UseNameOf
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.Editing;
+    using Microsoft.CodeAnalysis.Simplification;
+
+    [Shared]
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CSharpUseNameofCodeFixProvider))]
+    internal class CSharpUseNameofCodeFixProvider : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(
+            IDEDiagnosticIds.UseNameofDiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(
+                    new CodeAction.DocumentChangeAction(
+                        "Use nameof.",
+                        _ => CreateChangedDocument(diagnostic),
+                        "Use nameof."),
+                    diagnostic);
+            }
+
+            return Task.CompletedTask;
+
+            async Task<Document> CreateChangedDocument(Diagnostic diagnostic)
+            {
+                var syntaxRoot = await context.Document
+                                              .GetSyntaxRootAsync(context.CancellationToken)
```

Comment:
this is the wrong cancellation token to use.   you need the one that was passed (but ignored) in ```_ => CreateChangedDocument(diagnostic),```.

Reply:
Derp, dumbed that one.